crates/router/src/core/payments/transformers.rs

```diff
@@ -1707,12 +1707,30 @@ impl<F: Clone> TryFrom<PaymentAdditionalData<'_, F>> for types::PaymentsAuthoriz
                 None
             }
         });
-        let amount = payment_data
+        let mut amount = payment_data
             .surcharge_details
             .as_ref()
             .map(|surcharge_details| surcharge_details.final_amount)
             .unwrap_or(payment_data.amount.into());
 
+        let order_tax_amount = payment_data
+            .payment_intent
+            .tax_details
+            .clone()
+            .and_then(|tax| {
+                tax.payment_method_type
+                    .map(|a| a.order_tax_amount)
+                    .or_else(|| tax.default.map(|a| a.order_tax_amount))
```

Comment:
Minor refactor to slightly simplify this logic. Optional change by the way, you can choose to not take it up.

```suggestion
                tax.payment_method_type
                    .or(tax.default)
                    .map(|a| a.order_tax_amount)
```